flow/activities/flowable.go

```diff
@@ -297,16 +304,56 @@ func (a *FlowableActivity) MaintainPull(
 	}
 
 	done := make(chan struct{})
+	normalize := make(chan NormalizeBatchRequest)
+	normalizeDone := make(chan struct{})
 	a.CdcCacheRw.Lock()
 	a.CdcCache[sessionID] = CdcCacheEntry{
-		connector: srcConn,
-		done:      done,
+		connector:     srcConn,
+		done:          done,
+		normalize:     normalize,
+		normalizeDone: normalizeDone,
 	}
 	a.CdcCacheRw.Unlock()
 
 	ticker := time.NewTicker(15 * time.Second)
 	defer ticker.Stop()
 
+	go func() {
+	loop:
+		for {
+			select {
+			case req, ok := <-normalize:
+				if !ok {
+					break loop
+				}
+			retry:
+				if err := a.StartNormalize(ctx, config, req.BatchID); err != nil {
+					a.Alerter.LogFlowError(ctx, config.FlowJobName, err)
+					for {
+						// update req to latest normalize request & retry
+						select {
+						case req = <-normalize:
+						case <-done:
+							break loop
+						case <-ctx.Done():
+							break loop
+						default:
+							time.Sleep(time.Second)
```

Comment:
or change to 30 seconds for retry.

Reply:
at 30s I want to select on ctx.Done/syncDone to avoid delays reacting to pause, but leaving it since having this dance a third time would be a bit of a mess. Maybe something like shared.Sleep(context.Context) could work, but that won't watch over syncDone